### PR TITLE
Check that circuit execution time is shorter than `repetition_duration`

### PR DESCRIPTION
**Pull request by visagim**
As title says

### CODE REVIEW

File: src/qililab/platform/platform.py [hunk 1]
```diff
@@ -655,6 +658,11 @@ def compile(self, program: PulseSchedule | Circuit, num_avg: int, repetition_dur
 
         programs = {}
         for pulse_bus_schedule in pulse_schedule.elements:
+            if len(pulse_bus_schedule.timeline) != 0:  # can't do only one conditional if list is empty
```
Comment (AlbertMitjans) [inline src/qililab/platform/platform.py:661]: Will the list ever be empty??

Comment (AlbertMitjans) [reply src/qililab/platform/platform.py:661]: If so, shouldn't we list an error when it is empty?

Comment (visagim) [reply src/qililab/platform/platform.py:661]: it's empty for the flux schedules if there are no flux pulses. I believe we haven't fixed that yet.

File: docs/releases/changelog-dev.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ### New features since last release
 
+- Raise an error is a user requests to run a circuit that is longer than `repetition_duration`
```
Comment (AlbertMitjans) [inline docs/releases/changelog-dev.md:5]:
```suggestion
- Raise an error if a user requests to run a circuit that is longer than `repetition_duration`
```

**Review by AlbertMitjans — APPROVED** [review]
Good job! Just one small suggestion.

**Pull request merged**